bundle/Command/ScheduledVisibilityUpdateCommand.php

```diff
@@ -0,0 +1,248 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use function count;
+use function sprintf;
+
+final class ScheduledVisibilityUpdateCommand extends Command
+{
+    private SymfonyStyle $style;
+    private array $languageCache = [];
+
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ScheduledVisibilityService $scheduledVisibilityService,
+        private readonly Configuration $configurationService,
+        private readonly Connection $connection,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'Updates content visibility based on publish_from and publish_to attributes and configuration.',
+        );
+        $this->addOption(
+            'limit',
+            'l',
+            InputOption::VALUE_OPTIONAL,
+            'Number of content object to process in a single iteration',
+            1024,
+        );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->style = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->style->info(
+            'This command fetches content and updates visibility based on its schedule from publish_from and publish_to fields.',
+        );
+
+        $question = new ConfirmationQuestion(
+            'Continue with this action?',
+            true,
+            '/^(y)/i',
+        );
+
+        if (!$this->style->askQuestion($question)) {
+            $this->style->success('Aborted');
+
+            return Command::SUCCESS;
+        }
+
+        if (!$this->configurationService->isEnabled()) {
+            $this->style->warning('Scheduled visibility mechanism is disabled.');
+
+            return Command::FAILURE;
+        }
+
+        $allContentTypes = $this->configurationService->isAllContentTypes();
+        $allowedContentTypes = $this->configurationService->getAllowedContentTypes();
+
+        if (!$allContentTypes && count($allowedContentTypes) === 0) {
+            $this->style->warning('No content types configured for scheduled visibility mechanism.');
+
+            return Command::FAILURE;
+        }
+
+        $pager = $this->getPager($allContentTypes, $allowedContentTypes);
+
+        if ($pager->getNbResults() === 0) {
+            $this->style->info('No content found');
+
+            return Command::FAILURE;
+        }
+
+        $limit = $input->getOption('limit');
+        $offset = 0;
+
+        $this->style->createProgressBar($pager->getNbResults());
```

Comment:
I think it would be useful to have more info here, for example time estimation and memory consumption. You can do that with:

```php
$progressBar = $this->style->createProgressBar($pager->getNbResults());
$progressBar->setFormat('debug');
```

Reply:
fixed in 3eabc9d3c53d03f56d2ed1d215a55f9be0b3d043

Reply:
After configuring progress bar as in the example above, you can still use `SymfonyStyle` to update it, as was the case before.

Reply:
As communicated, we can't use SymfonyStyle to update it